### ISSUE TITLE
Ignored Test causing Suite to be marked as Ignored rather than Pass/Fail/Inconclusive

### ISSUE THREAD

**Issue by Bbuff101** [nunit/nunit#2617]
Hello,

We've been looking to migrate from NUnit2 to NUnit3, but have noticed some differences in result output with ignored test cases and how they relate to the overall test suite result.

In NUnit2, we noticed that when a single test case within a suite was ignored, then the suite would still be marked as pass/fail depending on the other test cases within that suite.  If all the test cases within the suite were ignored, then the test suite would be marked as inconclusive.

In NUnit3, we noticed that when a single test case within a suite was ignored, then the suite would be marked as ignored/skipped.  If all the tests cases within the suite were ignored, then the test suite would be marked as ignored/skipped.

Is this expected behavior in NUnit3 when a single test case or all test cases are ignored within a suite?  Or, should the test suites still be marked as pass/fail/inconclusive?

I've attached sample code and the resulting XML.  The runners I used were 2.6.4 against framework 2.6.4 and 3.7.0 against framework 3.9.0.  The commands I used were as follows:

NUnit2:
`'C:\Program Files (x86)\NUnit 2.6.4\bin\nunit-console.exe' .\NUnit2IgnoreAllTests\bin\Debug\NUnit2IgnoreAllTests.dll /labels:All result:".\Results\NUnit2IgnoreAllTest.xml"`

NUnit3:
`.\packages\NUnit.ConsoleRunner.3.7.0\tools\nunit3-console.exe .\NUnit3IgnoreAllTests\
bin\Debug\NUnit3IgnoreAllTests.dll --labels=All --result=".\Results\NUnit3IgnoreAllTest.xml"`

I've attached the code I wrote along with the result XMLs.
[NUnitIgnore.zip](https://github.com/nunit/nunit/files/1556595/NUnitIgnore.zip)

**Comment by CharliePoole** [comment]
Originally, NUnit had only two boolean attribites in the XML to use in representing the result: `executed` and `success`. In some of the last releases of NUnit V2 the `result` attribute was added to give a bit more information to those who needed it. This worked well for test cases, which is all most people producing reports were looking at. In the case of Suites, the result did not propogate from the contained test cases because usually the suites were executed, making a result like `Ignored` a little strange.

Oddly, however, we did propagate the result to the top level, even though it isn't shown in the XML. Both the console runner and the GUI show the overall result as Ignored if there are any ignored tests. The GUI colors all parent tests yellow if there are any ignored children.

In other words, we treated ignoring as "not running" in the XML but as a type of warning in the UI!

In NUnit 3 we decided to always treat ignored tests as a warning. When a suite is marked as Ignored, it may mean that it was ignored itself or simply that one child was ignored. We felt this was justified since there is added information in the element representing the suite that tells you how many tests passed, failed, were skipped, etc.

If you are looking at the XML, this might be considered a breaking change with NUnit 3. We didn't call it that because the entire structure of the XML was changing anyway.

**Comment by rprouse** [comment]
@Bbuff101 I am going to treat this issue as a question that I believe has been answered, so I am closing. If you are not happy with @CharliePoole's answer or have more questions, please post and we will continue the conversation.
